packages/app/src/cli/models/app/app.ts

```diff
@@ -1,22 +1,102 @@
 import {AppErrors} from './loader.js'
 import {ExtensionInstance} from '../extensions/extension-instance.js'
+import {isType} from '../../utilities/types.js'
 import {zod} from '@shopify/cli-kit/node/schema'
 import {DotEnvFile} from '@shopify/cli-kit/node/dot-env'
 import {getDependencies, PackageManager, readAndParsePackageJson} from '@shopify/cli-kit/node/node-package-manager'
 import {fileRealPath, findPathUp} from '@shopify/cli-kit/node/fs'
 import {joinPath, dirname} from '@shopify/cli-kit/node/path'
+import {AbortError} from '@shopify/cli-kit/node/error'
 
-export const AppConfigurationSchema = zod.object({
+export const LegacyAppSchema = zod.object({
+  extension_directories: zod.array(zod.string()).optional(),
+  web_directories: zod.array(zod.string()).optional(),
+  scopes: zod.string().default(''),
+})
+
+export const AppSchema = zod.object({
   extension_directories: zod.array(zod.string()).optional(),
   web_directories: zod.array(zod.string()).optional(),
   name: zod.string().optional(),
   client_id: zod.string().optional(),
```

Comment:
This would need to be required right?

Reply:
yeah this was example config, was gonna update in a separate pr but we can do that here